arch-install.bash

```diff
@@ -86,47 +90,106 @@ while true; do
   clear
 done
 
-sgdisk --zap-all "$block_device"
+if [[ "$encryption_choice" == "Yes" ]]; then
+  wipe_choice=$(printf 'Yes\nNo\n' | fzy -p 'do you want to secure wipe your disk?')
+  if [[ "$wipe_choice" == "Yes" ]]; then
+    drive_choice=$(printf 'SSD\nHDD\n' | fzy -p 'what kind of drive do you have?')
+    # Erase disk
+    case "$drive_choice" in 
+      #wont add nvme for now. No way to test
+      'SSD')
+        frozen_state=$(hdparm -I "$block_device" 2>/dev/null | awk '/frozen/ { print $1,$2 }')
+        if [ "${frozen_state}" == "not frozen" ]; then
+            hdparm --user-master u --security-set-pass password "$block_device"
+            hdparm --user-master u --security-erase password "$block_device"
+        else
+            echo "Your drive is frozen. Please fix!" >&2
+            exit 1 
+        fi
+      ;;
+      'HDD')
+        cryptsetup open --type plain -d /dev/urandom "$block_device" to_be_wiped
+        dd if=/dev/zero of=/dev/mapper/to_be_wiped status=progress bs=1M
+        cryptsetup close to_be_wiped
+      ;;
+    esac
+  fi
 
-case "$bios" in
-  'uefi')
-    # efi part
-    sgdisk --new=1:0:+512M "$block_device"
-    # root
-    sgdisk --new=2:0:0 "$block_device"
-    ;;
-  'legacy')
-    # root
-    parted "$block_device" mklabel msdos mkpart primary 0% 100%
-    ;;
-esac
+  sgdisk --zap-all "$block_device"
 
-partitions=()
-query=$(sfdisk --json "$block_device")
-for k in $(jq '.partitiontable.partitions | keys | .[]' <<< "$query"); do
-  partitions+=("$(jq --argjson k "$k" -r '.partitiontable.partitions | .[$k] | .node' <<< "$query")")
-done
+  sgdisk --new=1:0:+512M "$block_device"
+  sgdisk --new=2:0:0 "$block_device"
 
-case "$bios" in
-  'uefi')
-    efi_fs="${partitions[0]}"
-    root_fs="${partitions[1]}"
-    umount --quiet "$efi_fs" "$root_fs" || true
-    mkfs.fat -F32 "$efi_fs"
-    mkfs.ext4 -F "$root_fs"
+  partitions=()
+  query=$(sfdisk --json "$block_device")
+  for k in $(jq '.partitiontable.partitions | keys | .[]' <<< "$query"); do
+    partitions+=("$(jq --argjson k "$k" -r '.partitiontable.partitions | .[$k] | .node' <<< "$query")")
+  done
+
+  echo "Setting up Encryption using Cryptsetup"
+
+  cryptsetup -y -v  luksFormat --type luks1 "${partitions[1]}"
```

Comment:
We might as well use `--type luks2`. At least it works on my machine without issues.

Reply:
Fixed. luks2 is implicitly set